### PR TITLE
machine_learning: change Kconfig.sysbuild to support nRF53 DFU

### PR DESCRIPTION
**Pull request by sigvartmh**
Seems not all sysbuild configurations needed was enabled for this sample.

Fixes: NCSDK-28070

### CODE REVIEW

**Review by nordicjm — CHANGES_REQUESTED** [review]
this is wrong, the options are selected in https://github.com/nrfconnect/sdk-nrf/blob/main/applications/machine_learning/Kconfig.sysbuild

You can't select netcore secure boot on nrf52 devices

File: applications/machine_learning/Kconfig.sysbuild
Comment (sigvartmh) [inline applications/machine_learning/Kconfig.sysbuild:23]: @nordicjm so how do you default choices for boards? because this didn't work and making it a choice didn't seem to do what I intended.

Comment (nordicjm) [reply applications/machine_learning/Kconfig.sysbuild:23]: it needs to be in the choice part i.e.
```
choice MCUBOOT_MODE
default MCUBOOT_MODE_OVERWRITE_ONLY if SOC_SERIES_NRF53
endchoice
```

**Review by dchat-nordic — CHANGES_REQUESTED** [review]
The commit name is no longer valid. Please change

File: applications/machine_learning/Kconfig.sysbuild
Comment (sigvartmh) [reply applications/machine_learning/Kconfig.sysbuild:23]: Well as I tried to explain above I've tried this:

```
warning: default on the choice symbol MCUBOOT_MODE_OVERWRITE_ONLY (defined at /applications/machine_learning/Kconfig.sysbuild:23, /nrf/sysbuild/Kconfig.mcuboot:42) will have no effect, as defaults do not affect choice symbols
```

I can make a change directly to the Kconfig but that doesn't make sense either.

Comment (nordicjm) [reply applications/machine_learning/Kconfig.sysbuild:23]: I don't know what you mean, I just put this in `Kconfig.sysbuild` and it works perfectly??
```
choice MCUBOOT_MODE
        default MCUBOOT_MODE_OVERWRITE_ONLY if SOC_SERIES_NRF53X
endchoice
```

Comment (sigvartmh) [reply applications/machine_learning/Kconfig.sysbuild:27]: I think I was just stupid.

**Review by nordicjm — APPROVED** [review]

**Review by dchat-nordic — CHANGES_REQUESTED** [review]
The commit name is not valid. 
The commit says about sysbuild.conf, but the change is Kconfig.sysbuild.
Please change

**Review by nvlsianpu — APPROVED** [review]

**Comment by nvlsianpu** [comment]
commit message might mention that:
 this fix configuration for nRF5340 cpuapp targets so it uses overwrite mode by default which imply CONFIG_NRF53_MULT_IMAGE_UPDATE set as well. This makes default configuration matches to how simultaneous multi-image update is supported.

**Review by dchat-nordic — APPROVED** [review]

**Comment by nvlsianpu** [comment]
@sigvartmh Rebase?

**Review by jumi-nordic — APPROVED** [review]

**Comment by shanthanordic** [comment]
Dont merge until we get +1 from @grochu

**Review by grochu — APPROVED** [review]
nRF5340 update works fine.

**Comment by shanthanordic** [comment]
removed DNM as @grochu  approved.

**Comment by nvlsianpu** [comment]
CI issue is unrelated to PR content.
Other sample (sample.ipc.ipc_service.nrf5340dk_rpmsg_default) gets -D <empty> argument in command line building on a test-repo.

**Pull request merged**
